VisualPinball.Unity/VisualPinball.Unity/Extensions/MaterialExtensions.cs

```diff
@@ -3,144 +3,75 @@
 	{
-		private static readonly Logger Logger = LogManager.GetCurrentClassLogger();
-
-		private static readonly int Color = Shader.PropertyToID("_Color");
-		private static readonly int Metallic = Shader.PropertyToID("_Metallic");
-		private static readonly int Glossiness = Shader.PropertyToID("_Glossiness");
-		private static readonly int Mode = Shader.PropertyToID("_Mode");
-		private static readonly int SrcBlend = Shader.PropertyToID("_SrcBlend");
-		private static readonly int DstBlend = Shader.PropertyToID("_DstBlend");
-		private static readonly int ZWrite = Shader.PropertyToID("_ZWrite");
-		private static readonly int MainTex = Shader.PropertyToID("_MainTex");
-		private static readonly int BumpMap = Shader.PropertyToID("_BumpMap");
-
-		public static UnityEngine.Material ToUnityMaterial(this PbrMaterial vpxMaterial, TableBehavior table, StringBuilder debug = null)
-		{
-			if (table != null) {
-				var existingMat = table.GetMaterial(vpxMaterial);
-				if (existingMat != null) {
-					return existingMat;
-				}
-			}
 
-			var unityMaterial = new UnityEngine.Material(GetShader()) {
-				name = vpxMaterial.Id
-			};
-
-			// apply some basic manipulations to the color. this just makes very
-			// very white colors be clipped to 0.8204 aka 204/255 is 0.8
-			// this is to give room to lighting values. so there is more modulation
-			// of brighter colors when being lit without blow outs too soon.
-			var col = vpxMaterial.Color.ToUnityColor();
-			if (vpxMaterial.Color.IsGray() && col.grayscale > 0.8) {
-				debug?.AppendLine("Color manipulation performed, brightness reduced.");
-				col.r = col.g = col.b = 0.8f;
-			}
-			unityMaterial.SetColor(Color, col);
-
-			// validate IsMetal. if true, set the metallic value.
-			// found VPX authors setting metallic as well as translucent at the
-			// same time, which does not render correctly in unity so we have
-			// to check if this value is true and also if opacity <= 1.
-			if (vpxMaterial.IsMetal && (!vpxMaterial.IsOpacityActive || vpxMaterial.Opacity >= 1)) {
-				unityMaterial.SetFloat(Metallic, 1f);
-				debug?.AppendLine("Metallic set to 1.");
-			}
+		private static readonly Logger Logger = LogManager.GetCurrentClassLogger();
 
-			// roughness / glossiness
-			unityMaterial.SetFloat(Glossiness, vpxMaterial.Roughness);
+		/// <summary>
+		/// Lazy instantiation of the material converter
+		/// </summary>
+		private static readonly Lazy<IMaterialConverter> lazyMaterialConverterInstance = new Lazy<IMaterialConverter>(() => CreateMaterialConverter());
 
-			// blend mode
-			ApplyBlendMode(unityMaterial, vpxMaterial.MapBlendMode);
-			if (vpxMaterial.MapBlendMode == BlendMode.Translucent) {
-				col.a = Mathf.Min(1, Mathf.Max(0, vpxMaterial.Opacity));
-				unityMaterial.SetColor(Color, col);
-			}
+		/// <summary>
+		/// Singleton Material Converter
+		/// </summary>
+		public static IMaterialConverter MaterialConverter { get { return lazyMaterialConverterInstance.Value; } }

```

Comment:
We've been preferring the => shorthand whenever possible

Reply:
Thanks, I updated the code.